File: day2/day2.swift
let a = CommandLine.arguments[1].split(separator: "\n")
var sum = 0;
for l in a {
    var max = 0;
    var min = Int.max
    for z in l.split(separator:"\t") {
        let i = Int(z)!
        if i > max {
            max = i
        }
        if i < min {
            min = i
        }
    }
    sum += max-min
}
print(sum)

sum = 0;
for l in a {
    let s = l.split(separator:"\t")
    outerLoop:  for z in s {
        let i = Int(z)!
        for y in s {
            let j = Int(y)!
            if i==j {
                continue
            }
            if i%j == 0 {
                sum += i/j
                break outerLoop
            }
        }
    }
}
print(sum)

File: day2/day2.py
import sys,itertools
l=sys.argv[1].split("\n")
print(sum([max(k)-min(k)for k in[[int(j)for j in i.split("\t")]for i in l]]))
print(sum([[b[0]/b[1]for b in itertools.permutations([int(j)for j in i.split("\t")],2)if b[0]%b[1]==0][0]for i in l]))
